### DISCUSSION TITLE
Question about Network Queues

### DISCUSSION THREAD

**Discussion by nickgabbard** [smartin015/continuousprint#103] [Q&A] [answered]
I'm having trouble getting network queues to work.  When you put in an IP address and port number, what needs to be running at that address and port number?  

For testing, I've got 2 docker containers each running octoprint with the plugins.  They were created in the same docker-compose file on the same network.  The have 172.28.0.2 and 172.28.0.3 ip addresses respectively.  What do I need to put in each instance of the continuous print settings to make this work.  Everything I seem to put only shows 1 printer (itself) as part of that queue.

EDIT -- I would like to point out that the default auto queue (now that I've installed the RC and cleared the plugin data) does work just fine.  I'm now just more curious about what values were supposed to be there.

**Comment by smartin015** [comment, marked as answer]
Hi Nick - thanks for trying out the network queues! Glad you managed to get them to work, with the defaults in the RC no less.

The name of the lan queue must match between instances (consider it as a case-sensitive namespace - "LAN1" is a different queue entirely from "LAN2").

The address should be that insurance's IP address (or hostname, probably) suffixed with a valid, available port. That's used to bind the interface and port for the instance to serve/listen from.

Using your example, the docker host 172.28.0.2's config could look like:

Name "LAN", address "172.28.0.2:5678"

And the other would be:

Name "LAN", address "172.28.0.3:5678"

Note that, currently, changing addresses on a running insurance may require a reboot as I don't think it's rebound on change. If your IP changes frequently, the default queue IP assigned on DB init may also go stale... that's another quality improvement I've not yet gotten to.

Be advised that LAN queues are still pretty new, so if you have any FRs / issues I'd welcome a report so I can improve them. Thanks!

**Comment by nickgabbard** [comment]
I've been messing around with it, and it seems to be working like you expected it to.  

I know there is an option to automatically add a file to the queue on upload, but it seems to go to the local queue for that printer.  Am I missing an option somewhere to send it to the network queue instead?

Also, I've noticed that the auto-add to queue thing doesn't work on files that get dropped into the Watched folder in Octoprint.  Is that by design?

**Reply by smartin015** [reply]
Hey, thanks for your patience - just got back from two weeks of travel. 

There's not yet an option to submit directly to LAN - I want this to be possible after LAN printing becomes a bit more fluid and editable, like it is with the the local queue. I plan to look into this as part of #101.

TIL watched folders are a thing :) the auto-add behavior relies on events fired by the REST API handler, which captures user interaction via the UI and various slicers making HTTP calls. Apparently the file watcher just [calls the FileManager directly](https://github.com/OctoPrint/OctoPrint/blob/f430257d7072a83692fc2392c683ed8c97ae47b6/src/octoprint/server/util/watchdog.py#L119) without sending any events. This needs a patch to OctoPrint to have it send the proper events. I've opened #107 for this.
